src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -684,6 +684,66 @@ void Unit::DealDamageMods(Unit* victim, uint32 &damage, uint32* absorb)
             *absorb += damage;
         damage = 0;
     }
+
+    if (sWorld->getBoolConfig(CONFIG_LEGACY_BUFF_ENABLED))
+    {
+        int16 levelDiff = getLevel() - victim->getLevelForTarget(this);
+
+        if (levelDiff > 0)
+        {
+            const uint32 intendedItemLevelByExpansion[CURRENT_EXPANSION] = { 65, 115, 200, 346, 0, 0 };
+            float damageMultiplier = 1.0f;
+
+            // Damage increase with expansion/level difference
+            if (ToPlayer() && victim->ToCreature())
+            {
+                uint32 expansion = victim->ToCreature()->GetCreatureTemplate()->RequiredExpansion;
+
+                if (expansion < EXPANSION_MISTS_OF_PANDARIA && getLevel() > Trinity::GetMaxLevelForExpansion(expansion))
+                {
+                    if (levelDiff < 5)
+                        damageMultiplier = 1.0f + (0.0625f * levelDiff);
+                    else if (levelDiff < 10)
+                        damageMultiplier = 1.5f + (0.5f * levelDiff);
+                    else
+                        damageMultiplier = 16.5f;
+                }
+
+                uint32 maxLevelInExpansion = Trinity::GetMaxLevelForExpansion(expansion);
+                uint32 intendedItemLevelDiff = ToPlayer()->GetAverageItemLevel() - intendedItemLevelByExpansion[expansion];
```

Comment:
what happens when the character has no items on him? The intendedItemLevelDiff will then be `0-65` which becomes `4294967231`, which is probably not intended.

Btw. Noticed that GetAverageItemLevel seems to have a possible zero division.